### PR TITLE
throw if user passes a command-line arg

### PR DESCRIPTION
**Pull request by sethladd**

### CODE REVIEW

**Comment by sethladd** [comment]
Addresses https://github.com/dart-lang/dartdoc/issues/591

**Comment by sethladd** [comment]
cc @devoncarew

File: bin/dartdoc.dart [hunk 1]
```diff
@@ -19,7 +19,7 @@ main(List<String> arguments) async {
   var args = parser.parse(arguments);
 
   if (args['help']) {
-    _printUsageAndExit(parser);
+    _printHelp(parser);
```
Comment (devoncarew) [inline bin/dartdoc.dart]: will this cause docs to be generated, if we don't exit here?

Comment (devoncarew) [inline bin/dartdoc.dart]: nevermind

File: bin/dartdoc.dart [hunk 2]
```diff
@@ -78,6 +78,11 @@ main(List<String> arguments) async {
     packageRootDir = new Directory(_resolveTildePath(args['package-root']));
   }
 
+  if (args.rest.isNotEmpty) {
+    print('Error: detected unknown command-line argument(s)');
```
Comment (devoncarew) [inline bin/dartdoc.dart]: perhaps also: `: ${args.rest.join(' ')}`

**Comment by sethladd** [comment]
Nicer error message now.

also, s/warning/error since we halt the app if those situations are encountered.

**Comment by devoncarew** [comment]
lgtm

**Comment by sethladd** [comment]
Committed with d7397ab

**Pull request closed without merging**